test/e2e/marketplace_test.go

```diff
@@ -1,41 +1,67 @@
 package e2e
 
 import (
-	"context"
 	"fmt"
 	"testing"
 
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
 	"github.com/operator-framework/operator-marketplace/pkg/apis"
 	marketplace "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
-
-	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/test/helpers"
+	testsuites "github.com/operator-framework/operator-marketplace/test/testsuites"
 
 	"github.com/operator-framework/operator-sdk/pkg/test"
 
-	apps "k8s.io/api/apps/v1"
-	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-const (
-	GroupLabel string = "opsrc-group"
-)
-
-// Test marketplace is the root function that triggers the set of e2e tests
+// Test marketplace is the root function that triggers the set of e2e tests.
 func TestMarketplace(t *testing.T) {
-	// Add marketplace types to test framework scheme
+	initTestingFramework(t)
+
+	// init-operator-source creates an opeatorsource and then runs a a series of test suites that rely on this resource.
+	t.Run("init-operator-source", func(t *testing.T) {
+		// Create a ctx that is used to delete the operatorsource at the completion of this function.
+		ctx := test.NewTestCtx(t)
+		defer ctx.Cleanup()
+
+		// Get test namespace.
+		namespace, err := test.NewTestCtx(t).GetNamespace()
+		if err != nil {
+			t.Errorf("Could not get namespace: %v", err)
+		}
+
+		// Get global framework variables.
+		f := test.Global
+
+		// Create the operatorsource.
```

Comment:
It seems like the configuration bits could get very long if we had lots of tests. Maybe the configuration which is general to all of these tests (for example, getting the namespace would happen in every test) should happen in this function, and code which is custom setup per suite (like creating an `operatorsource` should be defined in each suite file?

Reply:
I'm glad that you brought this up @kevinrizza,

I'm not opposed to creating resources within in the test suites, but this means that resources can't be shared between test suites. If everyone thinks that the clarity outweighs the dip in performance brought on by recreating resources per test suite I am happy to make the change.

Reply:
No, I'm saying that you can still create things within the context of groups of tests. The calling of the tests and the resource generation of the tests are orthogonal, and it would be more readable if the resources created for each test group happened somewhere other than this method. In the same way, it might be nice to be able to create a hierarchy of tests more than one level deep (for example, a bunch of tests will all use the same `operatorsource`, and some sets of tests want to share the same `catalogsourceconfig` but not all). If that was a common pattern, having the top level resources created in this method doesn't make much sense.